scalding-core/src/main/scala/com/twitter/scalding/TypedDelimited.scala

```diff
@@ -47,15 +48,15 @@ trait TypedSeperatedFile extends Serializable {
 /**
  * Typed tab separated values file
  */
-object TypedTsv extends TypedSeperatedFile {
```

Comment:
So I'm thinking we could add a new macro (not implicit) in TypeDescriptorProviderImpl which is like caseClassTypeDescriptorCommonImpl, but which takes fields as a parameter and uses that value of fields, instead of using the helper macro.

We would then call this macro from the `apply` methods of TypedSeparatedFile which specify a fields parameter.